backend-docker/pier/db/db.go

```diff
@@ -0,0 +1,551 @@
+
+// VolumeID contains a docker volume ID
+type VolumeID dckr.VolumeID
+
+// Db is used to keep DbMap
+type Db struct{ gorp.DbMap }
+
+// Job stores the information about a service execution (used to serialize JSON)
+type Job struct {
+	ID           JobID
+	ServiceID    ServiceID
+	Input        string
+	Created      time.Time
+	State        *JobState
+	InputVolume  VolumeID
+	OutputVolume VolumeID
+	Tasks        []Task
+}
+
+// JobState keeps information about a job state
+type JobState struct {
+	Error  string
+	Status string
+	Code   int
+}
+
+// JobID exported
+type JobID string
+
+// JobTable stores the information about a service execution (used to store data in a database)
+type JobTable struct {
+	ID           string
+	ServiceID    string
+	Input        string
+	Created      time.Time
+	Error        string
+	Status       string
+	Code         int
+	InputVolume  string
+	OutputVolume string
+	Revision     int
+}
+
+// Task contains tasks related to a specific job (used to serialize JSON)
+type Task struct {
+	ID            string
+	Name          string
+	ContainerID   dckr.ContainerID
+	Error         string
+	ExitCode      int
+	ConsoleOutput string
+}
+
+// TaskTable contains tasks related to a specific job (used to store data in a database)
+type TaskTable struct {
+	ID            string
+	Name          string
+	ContainerID   string
+	Error         string
+	ExitCode      int
+	ConsoleOutput string
+	Revision      int
+	JobID         string
+}
+
+// LatestOutput used to serialize console output to JSON
+type LatestOutput struct {
+	Name          string
+	ConsoleOutput string
+}
+
+// Bind describes the binding between an IOPort and a docker volume
+type Bind struct {
+	IOPort   IOPort
+	VolumeID dckr.VolumeID
+}
+
+// Service describes metadata for a GEF service (used to serialize JSON)
+type Service struct {
+	ID          ServiceID
+	ImageID     dckr.ImageID
+	Name        string
+	RepoTag     string
+	Description string
+	Version     string
+	Created     time.Time
+	Size        int64
+	Input       []IOPort
+	Output      []IOPort
+}
+
+// ServiceTable describes metadata for a GEF service (used to store data in a database)
+type ServiceTable struct {
+	ID          string
+	ImageID     string
+	Name        string
+	RepoTag     string
+	Description string
+	Version     string
+	Revision    int
+	Created     time.Time
+	Size        int64
+}
+
+// ServiceID exported
+type ServiceID string
+
+// IOPort is an i/o specification for a service
+// The service can only read data from volumes and write to a single volume
+// Path specifies where the volumes are mounted
+type IOPort struct {
+	ID   string
+	Name string
+	Path string
+}
+
+// IOPortTable is used to store data in a database
+type IOPortTable struct {
+	ID        string
+	Name      string
+	Path      string
+	IsInput   bool
+	Revision  int
+	ServiceID string
+}
+
+// InitDb initializes the database engine
+func InitDb() (Db, error) {
+	var dataBaseHandler Db
```

Comment:
it's more difficult to trace the usage of this variable, and it's not really needed (use return Db{nil}, err or return Db{*dataBaseMap}, err)